In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#loadign dataset
from sklearn.datasets import load_iris

In [3]:
iris_dataset = load_iris()

In this dataset, we have 3 different iris species and measurements of these species. Let's have a look at the iris dataset to know what we have.

In [7]:
iris_dataset.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename'])

"target_names" and "feature_names" are contains the name of species and description of each feature.

In [9]:
iris_dataset["target_names"] #the 3 iris species names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

In [10]:
iris_dataset["feature_names"] #features names 

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

The actual data in "data" and "target" fields. "data" contains the numeric measurements of features and "target" contains species. "target" will be our dependent variable.

In [11]:
y = iris_dataset["target"]
x = iris_dataset["data"]

In [14]:
y # 0 -> setosa, 1 -> versicolor, 2 -> virginica

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [16]:
x[:5] #first 5 rows of measurements

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

Now we have to split these data into train and test, and scale x data.

In [17]:
from sklearn.model_selection import train_test_split

In [18]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.33, random_state=20)

In [21]:
from sklearn import preprocessing

In [22]:
scaler = preprocessing.MinMaxScaler()

In [23]:
X_train = scaler.fit_transform(x_train)
X_test = scaler.transform(x_test)

### SVM

In [24]:
from sklearn.svm import SVC

In [52]:
svm = SVC(kernel="rbf")

In [53]:
svm.fit(X_train, y_train)

SVC()

In [54]:
y_pred = svm.predict(X_test) #making predictions 

In [55]:
from sklearn.metrics import confusion_matrix

In [56]:
cm = confusion_matrix(y_test, y_pred) #creating confusion matrix for evaluation

In [57]:
cm

array([[17,  0,  0],
       [ 0, 18,  0],
       [ 0,  4, 11]], dtype=int64)

In [58]:
#also we can calculate the test score
test_score = np.mean(y_pred == y_test)

In [59]:
test_score

0.92

For this model, the test set accuracy is about 0.92, which means we made the right
prediction for 92% of the irises in the test set.